### ISSUE TITLE
Why are masked (protected) regions being altered during inpainting?

### ISSUE THREAD

**Issue by XuwuChen443** [bytedance/OneReward#10]
Hello!
I'm currently using the FluxFillCFGPipeline for an inpainting task and have observed some behavior that I don't fully understand. 
My specific use case involves repainting a very large portion of an image while preserving only a small, specific area. I am providing a binary mask where the black area corresponds to the content I want to keep untouched.
However, in the final output, I've noticed that the supposedly protected region is being modified. 
My core question is: Is this expected behavior? I was under the impression that the model should not alter the pixels in the masked-off (black) region at all. Perhaps I am missing a particular setting or there's a best practice for ensuring perfect preservation of protected areas?
Here the code:
```python
import torch
from diffusers.utils import load_image
from diffusers import FluxTransformer2DModel
from src.pipeline_flux_fill_with_cfg import FluxFillCFGPipeline
from PIL import Image
import math

def resize_for_model(image: Image.Image, div: int = 16):
    w, h = image.size
    
    new_h = math.ceil(h / div) * div
    new_w = math.ceil(w / div) * div
    
    if image.mode == 'L' or image.mode == '1': # Assuming mask is grayscale or binary
        resized_image = image.resize((new_w, new_h), Image.NEAREST)
        print(f"Resized MASK from ({w}, {h}) to ({new_w}, {new_h}) using NEAREST")
    else:
        resized_image = image.resize((new_w, new_h), Image.LANCZOS)
        print(f"Resized IMAGE from ({w}, {h}) to ({new_w}, {new_h}) using LANCZOS")
        
    return resized_image

transformer_onereward_dynamic = FluxTransformer2DModel.from_pretrained(
    ",
    subfolder="flux.1-fill-dev-OneRewardDynamic-transformer",
    torch_dtype=torch.bfloat16
)

pipe = FluxFillCFGPipeline.from_pretrained(
    "", 
    transformer=transformer_onereward_dynamic,
    torch_dtype=torch.bfloat16).to("cuda")

image_orig = load_image()
mask_orig = load_image()

image = resize_for_model(image_orig, 16)
mask = resize_for_model(mask_orig, 16)
print(image.height)
print(image.width)
result_image = pipe(
    prompt=',
    negative_prompt="nsfw",
    image=image,
    mask_image=mask,
    height=image.height,
    width=image.width,
    guidance_scale=1.0,
    true_cfg=4.0,
    num_inference_steps=50,
    generator=torch.Generator("cpu").manual_seed(0)
).images[0]
result_image.save(f".jpg")
```

**Comment by FreyWang** [comment]
The error comes from the training of DIT and VAE reconstruction

**Comment by XuwuChen443** [comment]
@FreyWang Is there any way i can do to resolve this?

**Comment by FreyWang** [comment]
resize the image height and width in range (1024, 2048) will be better
